code/app/global_variables.py

```diff
@@ -1,201 +1,72 @@
-    declared as global variables in the program. The global variables being used
-    in the program are listed below.
-    
-    x_nodes         : af.Array [2 1 1 1]
-                      Stores the domain of the wave. Default value is set to
-                      [-1, 1]. This means that the domain is from
-                      :math:`x \\epsilon [-1., 1.]`. This domain will be split
-                      into `Number_of_elements` elements.
-    
-    N_Elements      : int
-                      The number of elements into which the domain is to be
-                      divided.
-                 
-    N_LGL           : int
-                      The number of `LGL` nodes in an element.
-              
-    xi_LGL          : af.Array [N_LGL 1 1 1]
-                      `N_LGL` LGL points.
-    
-    element_LGL     : af.Array [N_LGL N_Elements 1 1]
-                      The domain is divided into `N_elements` number of elements
-                      and each element has N_LGL points corresponding to the
-                      each LGL point. This variable stores the :math:`x`
-                      coordinates corresponding to each `xi_LGL` for each
-                      element. element_LGL[n_LGL, n_element] returns the
-                      :math:`n\\_LGL^{th}` :math:`x` coordinate for the
-                      :math:`n_element^{th}` element.
-                  
-    lBasisArray     : af.Array [N_LGL N_LGL 1 1]
-                      Contains the coefficients of the Lagrange basis functions
-                      created using LGL nodes stored in xi_LGL.
-                      lBasisArray[i, j] is the coefficient corresponding to the
-                      :math:`(N_LGL - j - 1)^{th}` power of :math:`x` in the
-                      :mat:`i^th` Lagrange polynomial.
-    
-    dLp_xi          : af.Array [N_LGL N_LGL 1 1]
-                      Stores the value of the derivative
-                      :math:`\\frac{dL_p}{d\\xi}` at all the LGL points.
-    
-    lobatto_weights : af.Array [N_LGL 1 1 1]
-                      Lobatto weights to integrate using Gauss-Lobatto
-                      quadrature.
-                      
-    c               : float
-                      Number denoting the wave speed. Used in flux calculation.
-    
-    time            : af.Array
-                      Array containing the time for which the :math:`u` is to be
-                      calculated.
-    
-    u               : af.array [N_LGL N_Elements time.shape[0] 1]
-                      Stores :math:`u` calculated at the :math:`x` cordinates
-                      stored in the variable `element_LGL` at time :math:`t`
-                      stored in stored in variable `time`.
-        
-        
+    Calculates and returns the LGL points for a given N, These are roots of
+    the formula.
+    :math: `(1 - xi ** 2) P_{n - 1}'(xi) = 0`
+    Legendre polynomials satisfy the recurrence relation
```

Comment:
The variable `legendre_N_minus_1` uses this recurrence relation.

Reply:
It would have made sense to give the recurrence relation if you had explained the algorithm. But since you have not explained the algorithm, the reader won't know why you have written the recurrence relation.